fetch.js

```diff
@@ -225,9 +225,14 @@
     this.mode = options.mode || null
     this.referrer = null
 
+    if (!options.constructedWithFetch && methods.indexOf(this.method) === -1) {
```

Comment:
This consults `methods` array which is not a definite list of all supported HTTP methods, but only of few main methods that the spec defines as case-insensitive.

To properly implement this per-spec, we would have to maintain the [list of forbidden methods instead](https://fetch.spec.whatwg.org/#forbidden-method):

> A forbidden method is a method that is a byte-case-insensitive match for one of `CONNECT`, `TRACE`, and `TRACK`.

Furthermore, a ["method" is defined](https://tools.ietf.org/html/rfc7230#section-3.2.6) simply as a "token", which to me seems like it's basically any non-empty string.

So, we would have to check if the method is nonempty and if it's not listed among forbidden methods.